### PR TITLE
AWS::Logs::ResourcePolicy

### PR DESCRIPTION
**Pull request by jerry153fish**

### CODE REVIEW

File: troposphere/logs.py [hunk 1]
```diff
@@ -1,6 +1,21 @@
 from . import AWSObject, AWSProperty
 from .constants import LOGS_ALLOWED_RETENTION_DAYS as RETENTION_DAYS
-from .validators import integer_list_item
+from .validators import integer_list_item, json_checker
+
+
+def validate_resource_policy(policy_document):
```
Comment (markpeek) [inline troposphere/logs.py:6]: This validation should take into account the other policy document types coming from awacs. This is imported via https://github.com/cloudtools/troposphere/blob/main/troposphere/compat.py
Note: the length checks won't work with some of these awacs types and additional checking needs to be done around the json_checker return types.

Comment (jerry153fish) [reply troposphere/logs.py:6]: hi @markpeek So the plan will be

1. build a more general policy document validator
2. cover the types of dict, Policy, PolicyDocumen, and string
3. add tests cover all above

Any other places need to be considered here?

Comment (markpeek) [reply troposphere/logs.py:6]: That looks like the right list of items. Thanks.

**Comment by jerry153fish** [comment]
@markpeek I can not reproduce the flake errors on my mac and ubuntu locally, do you know what might happen?

**Comment by markpeek** [comment]
Excellent. Thank you for the commit.

**Comment by jerry153fish** [comment]
Thanks my pleasure.

**Comment by markpeek** [comment]
Oh good catch. Fixed via a7716408e1dc3693ec4157ea170eb6213ef85b36

```
    Fix duplicate AWS::Logs::ResourcePolicy

    The newly added LogResourcePolicy already existed as ResourcePolicy
    in Logs. Prefer the older one as it aligns with the AWS naming.
    This does keep the newly added validation and tests.
```

**Comment by jerry153fish** [comment]
thx, I missed that.

**Pull request merged**